### PR TITLE
Frontend Columns dnd

### PR DESCRIPTION
**Pull request by ChiuMungZitAlexander**
As backend $api is still in progress, so please just review. Or you can approve the request and after I will modify the $api ASAP. Thanks a lot😜😜😜😜😜

### CODE REVIEW

**Comment by beltekylevi** [comment]
please move the story in jira as well, also provide the task id here thanks!

File: src/client/components/boardColumn/index.js [hunk 1]
```diff
@@ -129,8 +168,18 @@ class List extends React.Component {
     }
 
     return (
-      <div className="board-column-wrapper">
-        <div className="board-column">
+      <div className="board-column-wrapper"
+        onDragEnter={this.handleDragEnter.bind(this)}
```
Comment (beltekylevi) [inline src/client/components/boardColumn/index.js]: you could bind `this` in the constructor

File: src/client/components/boardColumn/index.js [hunk 2]
```diff
@@ -129,8 +168,18 @@ class List extends React.Component {
...
+        onDragOver={this.handleDragOver.bind(this, this.props.column._id)}
```
Comment (beltekylevi) [inline src/client/components/boardColumn/index.js]: you could bind `this` in the constructor

File: src/client/components/boardColumn/index.js [hunk 3]
```diff
@@ -129,8 +168,18 @@ class List extends React.Component {
...
+        onDragLeave={this.handleDragLeave.bind(this)}
```
Comment (beltekylevi) [inline src/client/components/boardColumn/index.js]: you could bind `this` in the constructor

File: src/client/components/boardColumn/index.js [hunk 4]
```diff
@@ -129,8 +168,18 @@ class List extends React.Component {
...
+        onDrop={this.handleDrop.bind(this)}
```
Comment (beltekylevi) [inline src/client/components/boardColumn/index.js]: you could bind `this` in the constructor

File: src/client/components/boardColumn/index.js [hunk 5]
```diff
@@ -129,8 +168,18 @@ class List extends React.Component {
...
+        ref={(elem) => {
+          this.wrapper = elem;
+        }}>
+        <div className="board-column" draggable="true"
+          onDragStart={this.handleDragStart.bind(this, this.props.column._id)}
```
Comment (beltekylevi) [inline src/client/components/boardColumn/index.js]: you could bind `this` in the constructor

File: src/client/components/boardColumn/index.js [hunk 6]
```diff
@@ -129,8 +168,18 @@ class List extends React.Component {
...
+          onDrag={this.handleDrag.bind(this)}
```
Comment (beltekylevi) [inline src/client/components/boardColumn/index.js]: you could bind `this` in the constructor

File: src/client/components/boardColumn/index.js [hunk 7]
```diff
@@ -129,8 +168,18 @@ class List extends React.Component {
...
+          onDragEnd={this.handleDragEnd.bind(this)}>
```
Comment (beltekylevi) [inline src/client/components/boardColumn/index.js]: you could bind `this` in the constructor

File: src/client/components/boardDetail/index.js [hunk 8]
```diff
@@ -87,7 +118,8 @@ class BoardDetail extends React.Component {
       this.state.data.columns.forEach(function(element) {
         boardDisplay.push(
           <BoardColumn column={element} key={element._id}
-            deleteColumn={this.deleteColumn} addCard={this.addCard} />
+            deleteColumn={this.deleteColumn} addCard={this.addCard}
+            reorderColumns={this.reorderColumns.bind(this)} />
```
Comment (beltekylevi) [inline src/client/components/boardDetail/index.js]: you could bind `this` in the constructor

**Review by beltekylevi — CHANGES_REQUESTED** [review]

**Review by beltekylevi — APPROVED** [review]

**Comment by beltekylevi** [comment]
which story is this in jira?

**Comment by ChiuMungZitAlexander** [comment]
it is blp-71 drag and drop columns frontend

**Pull request merged**